src/main/java/com/hms/flexyosisoftconnector/dataserver/OSIsoftServerResponseUtil.java

```diff
@@ -0,0 +1,427 @@
+      if (response.has(MESSAGES_FIELD_KEY)) {
+        foundMessage = true;
+        messageSuccessful = handleMessages(response, connectionUrl, responseFileName);
+      }
+    } catch (Exception e) {
+      messageSuccessful = false;
+      Logger.LOG_SERIOUS(
+          "An error occurred while parsing the response for request: " + connectionUrl);
+      Logger.LOG_EXCEPTION(e);
+      if (foundMessage) {
+        Logger.LOG_SERIOUS(
+            "Message(s) such as a log, warning, or error, were received from the server, "
+                + "but the response is invalid.");
+      }
+    }
+
+    // Return boolean indicating message success (or failure)
+    return messageSuccessful;
+  }
+
+  /**
+   * Handles a single message in an HTTPS response from the OSIsoft server.
+   *
+   * @param response The response from the OSIsoft server to handle
+   * @param connectionUrl The URL of the OSIsoft server request that was made
+   * @throws Exception if an error occurs while parsing the response
+   */
+  public static void handleMessage(JSONObject response, String connectionUrl) throws Exception {
+    String message = response.getString(MESSAGE_FIELD_KEY);
+    Logger.LOG_SERIOUS(
+        "The following response message was received for request "
+            + connectionUrl
+            + ": "
+            + message);
+    Logger.LOG_SERIOUS(
+        "Check that the configured credentials/authentication information is correct.");
+  }
+
+  /**
+   * Handles an errors array in an HTTPS response from the OSIsoft server.
+   *
+   * @param response The response from the OSIsoft server to handle
+   * @param connectionUrl The URL of the OSIsoft server request that was made
+   * @throws Exception if an error occurs while parsing the response
+   */
+  public static void handleErrors(JSONObject response, String connectionUrl) throws Exception {
+    // Loop through each error and log it
+    JSONArray errors = response.getJSONArray(ERRORS_FIELD_KEY);
+    for (int i = 0; i < errors.length(); i++) {
+      String error = errors.getString(i);
+      Logger.LOG_SERIOUS(
+          "The following response error message was received for request "
+              + connectionUrl
+              + ": "
+              + error);
+
+      // Check for Web ID error
+      if (error.startsWith(OSIsoftServer.WEB_ID_ERROR_STRING)) {
+        Logger.LOG_SERIOUS("WEB ID: \"" + OSIsoftConfig.getServerWebID() + "\"");
+        Logger.LOG_SERIOUS("WEB ID Error: Supplied Web ID does not exist on this server");
+      }
+    }
+  }
+
+  /**
+   * Handles a messages array in an HTTPS response from the OSIsoft server.
+   *
+   * @param response The response from the OSIsoft server to handle
+   * @param connectionUrl The URL of the OSIsoft server request that was made
+   * @param responseFileName The name of the file which the response was saved to
+   * @return boolean indicating if an error severity message was found. If no error severity message
```

Comment:
The response value and variable name seem strange. 

"@return boolean indicating if an error severity message was found." leads you to initially thing true = error.

also why store an inverted boolean? (noErrorSeverityFound) when its much more typical to store a positive value [isErrorSeverityFound].

This inversion just seems strange and prone to user error.

Reply:
Updated Javadoc and inverted the boolean to `foundMessageWithErrorSeverity`